sunxi: add support for FriendlyARM NanoPi R1S-H5

maurerr opened the pull request:
It has the following features:

- Allwinner H5, Quad-core Cortex-A53
- 512MB DDR3 RAM
- 10/100/1000M Ethernet x 2
- RTL8189ETV WiFi 802.11b/g/n
- USB 2.0 host port (A)
- MicroSD Slot
- Serial Debug Port
- 5V 2A DC power-supply

Installation:

- Write the image to SD Card with dd
- Boot NanoPi from the SD Card

backported from https://github.com/torvalds/linux/commit/9962cb9be2db877c232aaf00db40125c0d7bf4bc

Signed-off-by: Marius Durbaca <mariusd84@gmail.com>

adschm commented:
This needs to be cleaned up. More merge commits than content ...

adschm commented:
https://openwrt.org/docs/guide-developer/frequent-pr-mistakes#do_not_multi-post_or_re-post_prs

maurerr commented:
really sorry for the multi-post - I'm just starting with github MRs. It should be cleaned up now

hauke commented on package/boot/uboot-sunxi/Makefile:
```diff
@@ -357,6 +366,7 @@ UBOOT_TARGETS := \
 	nanopi_neo_plus2 \
 	nanopi_neo2 \
 	nanopi_r1 \
+        nanopi_r1s_h5 \
```
Please use tabs for indention here

maurerr replied:
is it ok now ?

hauke replied:
No this is still using spaces instated of tabs.

hauke commented on package/boot/uboot-sunxi/Makefile:
```diff
@@ -270,6 +270,15 @@ define U-Boot/nanopi_neo2
 
+define U-Boot/nanopi_r1s_h5
+  BUILD_SUBTARGET:=cortexa53
+  NAME:=U-Boot for NanoPi R1S (H5)
```
Please remove the prefix `U-Boot for` here.

maurerr replied:
done

hauke commented on package/boot/uboot-sunxi/patches/253-sunxi-h5-add-support-for-nanopi-r1s.patch:
```diff
@@ -0,0 +1,295 @@
++#include <dt-bindings/input/input.h>
++
++/ {
++	model = "FriendlyARM NanoPi R1S H5";
```
Did you send this also to upstream U-boot to get it integrated there?

maurerr replied:
no, I is mandatory for this MR to get merged? as we donlt use latest in openwrt I thought it will get picked up later one when U-boot will sync with upstream linux as I notice they do about every year

hauke replied:
Should be ok, as it is already in upstream kernel.

maurerr replied:
alright. I'll send this also to u-boot upstream. They don't use github but mail patch only so a little challenging for me :) 
Anyway I hope we can move along wit this MR and not wait for u-boot upstream.

hauke commented on package/boot/uboot-sunxi/patches/253-sunxi-h5-add-support-for-nanopi-r1s.patch:
```diff
@@ -0,0 +1,295 @@
+ 	sun50i-h5-libretech-all-h5-cc.dtb \
+ 	sun50i-h5-nanopi-neo2.dtb \
+ 	sun50i-h5-nanopi-neo-plus2.dtb \
++    sun50i-h5-nanopi-r1s-h5.dtb \
```
Please use tabs here

maurerr replied:
done. is it ok now ?

hauke commented on target/linux/sunxi/image/cortexa53.mk:
```diff
@@ -40,6 +40,15 @@ define Device/friendlyarm_nanopi-neo2
+define Device/friendlyarm_nanopi-r1s-h5
+  DEVICE_VENDOR := FriendlyARM
+  DEVICE_MODEL := NanoPi R1S-H5
+  DEVICE_PACKAGES := kmod-rtc-sunxi kmod-usb-net-rtl8152 \
```
Is this RTC working fine on your board? Does it have a battery backup?

maurerr replied:
no it doesn't have an RTC so I've removed it

hauke commented on target/linux/sunxi/patches-5.4/302-sunxi-h5-add-support-for-nanopi-r1s.patch:
```diff
@@ -0,0 +1,296 @@
+--- a/Documentation/devicetree/bindings/arm/sunxi.yaml
++++ b/Documentation/devicetree/bindings/arm/sunxi.yaml
```
Please export  both patches from upstream Linux and add it with the commit header from here:
https://github.com/torvalds/linux/commit/9962cb9be2db877c232aaf00db40125c0d7bf4bc
https://github.com/torvalds/linux/commit/92ed3675574723a963152abbbe527b47f659340f
`git format-patch 9962cb9be2db877c232aaf00db40125c0d7bf4bc -1 `
`git format-patch 92ed3675574723a963152abbbe527b47f659340f -1 `

maurerr replied:
better now ?

hauke replied:
this looks good

maurerr replied:
please note that https://github.com/torvalds/linux/commit/9962cb9be2db877c232aaf00db40125c0d7bf4bc and https://github.com/torvalds/linux/commit/92ed3675574723a963152abbbe527b47f659340f doesn;t apply cleanly on 5.4 or .510 so I had to do some changes. I've kept the commit header as requested though

hauke reviewed (commented):
Please add the text from the pull request message also to the commit comment.

neheb commented on package/boot/uboot-sunxi/patches/253-sunxi-h5-add-support-for-nanopi-r1s.patch:
```diff
@@ -0,0 +1,295 @@
++		compatible = "gpio-leds";
++
++		led-0 {
++			label = "nanopi:green:lan";
```
label is deprecated upstream. Please switch to color and function.

adschm replied:
No, please just drop the model prefix and keep the label. We do not support color/function yet in OpenWrt.

https://github.com/openwrt/openwrt/pull/3437

maurerr replied:
fixed

adschm replied:
Err, I just realized that this is for uboot. So, probably nothing speaks against using color/function here, if you know how to use it properly.

In contrast, the relevant discussion would be what to do with the target patch. Is that one already upstreamed?
Yes -> Put the kernel version in the file name.
No -> One should consider to use label here. However, as long as no aliases are used, this will probably just work as expected with color/function.

adschm commented on target/linux/sunxi/base-files/etc/board.d/02_network:
```diff
@@ -10,6 +10,9 @@ case $(board_name) in
 friendlyarm,nanopi-r1)
 	ucidef_set_interfaces_lan_wan "eth1" "eth0"
 	;;
+friendlyarm,nanopi-r1s-h5)
```
This can be merged with previous block.

maurerr replied:
should be fixed now

hauke commented:
In commit 83672f506d95fe3d79b5afef6595e946ceeb4443 support for kernel 5.10 was added to the sunxi target, please add the patch also for kernel 5.10.

adschm commented on target/linux/sunxi/image/cortexa53.mk:
```diff
@@ -40,6 +40,15 @@ define Device/friendlyarm_nanopi-neo2
+  DEVICE_VENDOR := FriendlyARM
+  DEVICE_MODEL := NanoPi R1S-H5
+  DEVICE_PACKAGES := kmod-usb-net-rtl8152 kmod-leds-gpio \
+     kmod-ledtrig-heartbeat
```
Hanging indent with a single tab please.

maurerr committed:
sunxi: add support for FriendlyARM NanoPi R1S-H5

It has the following features:

- Allwinner H5, Quad-core Cortex-A53
- 512MB DDR3 RAM
- 10/100/1000M Ethernet x 2
- RTL8189ETV WiFi 802.11b/g/n (no driver in openwrt)
- USB 2.0 host port (A)
- MicroSD Slot
- Serial Debug Port
- 5V 2A DC power-supply

Installation:

- Write the image to SD Card with dd
- Boot NanoPi from the SD Card

backported from https://github.com/torvalds/linux/commit/9962cb9be2db877c232aaf00db40125c0d7bf4bc

Signed-off-by: maurerr <mariusd84@gmail.com>

hauke commented:
An other patch adding support for FriendlyARM NanoPi R1S-H5  was added in e43eb16efe97a597a2ebaa9f549d1daec2c8c2ab, fde68cb80941a60be93ece75e808b5b407d11cc8 and 1b18195f59b09844ab8fdfd669bc506c25df111b
